File: java/03-ArraysStringsHashTables/src/main/java/com/danielblanco/algoritmosestructuras/arraysstringshashtables/_01_is_unique/IsUnique.java
package com.danielblanco.algoritmosestructuras.arraysstringshashtables._01_is_unique;

import java.util.HashSet;
import java.util.Set;

/*
 * Dado un método que recibe una String, comprobar si todos los caracteres son únicos o no.
 *
 * isUnique("abcde") => true;
 * isUnique("abcded") => false;
 */
public class IsUnique {

  public boolean isUnique(String s) {
    if (s.length() > 128) return false;
     
    Set<Character> characters = new HashSet<>();
    for(Character c : s.toCharArray()) {
       if(characters.contains(c)) return false;
       characters.add(c);
    }
    
    return true;
  }
}

File: typescript/src/isUnique.ts
// Exercise 1: Java to TypeScript

/*
 * Dado un método que recibe una String, comprobar si todos los caracteres son únicos o no.
 *
 * isUnique("abcde") => true;
 * isUnique("abcded") => false;
 */

// TypeScript code:
export function isUnique(str: string): boolean {
  const strSet = new Set(str);
  return strSet.size === str.length;
}